go/protocol/keybase1/block.go

```diff
@@ -113,6 +113,22 @@ func (o PutBlockArg) DeepCopy() PutBlockArg {
 	}
 }
 
+type PutBlockAgainArg struct {
+	Folder   string         `codec:"folder" json:"folder"`
+	Ref      BlockReference `codec:"ref" json:"ref"`
```

Comment:
What do you mean by "both APIs"?  Do you mean you only want one `PutBlock` API call that can figure out which case we're in?  We could expand the arguments to `PutBlock` to include a reference, but that doesn't seem as clear as a separate API call to me, and not sure there's much of a cost to adding a new RPC.  (Plus we get the bonus of not worrying about backwards compatibility of old clients.)  But maybe I'm misunderstanding the suggestion.